graviti/dataframe/frame.py

```diff
@@ -627,12 +649,13 @@ def extend(self, values: Union[Iterable[Dict[str, Any]], "DataFrame"]) -> None:
         elif self.operations is not None:
             values = values.copy()
 
+        values_length = len(values)
```

Comment:
why we need `values_length`?

Reply:
if we use extend like `df.extend(df)`, then the `len(values)` will double